src/plugins/kernel_install/utils.sh

```diff
@@ -289,12 +289,28 @@ function list_installed_kernels()
   local prefix="$4"
   local -a available_kernels=()
   local cmd
+  local file_system_type
+  local ret=0
 
   # TODO: Drop me in the future
   migrate_old_kernel_list
 
-  cmd_manager "$flag" "sudo mkdir -p $REMOTE_KW_DEPLOY"
-  cmd_manager "$flag" "sudo touch $INSTALLED_KERNELS_PATH"
+  file_system_type=$(detect_filesystem_type '')
+  is_filesystem_writable "$file_system_type"
+  if [[ "$?" != 0 ]]; then
+    printf '%s\n' 'WARNING: /boot is read-only. Consider run: kw deploy --setup'
+    ret=30 # EROFS
+  fi
+
+  if [[ ! -f "${INSTALLED_KERNELS_PATH}" ]]; then
+    if [[ "$ret" == 30 ]]; then
+      printf '%s\n' 'No kernel managed by kw in a read-only /boot folder'
```

Comment:
The new approach to --list-all allows kw to list kernel installed in the system, even if none of them are managed by kw.
With this change, we cannot see those kernels anymore. I think we should warn users that "kw can't manage none of kernels in this system since it's a read-only /boot folder. Please consider run: kw deploy --setup" but continue listing the kernels installed in the system.